### PR TITLE
fix: adapt to new sjtu-jwc webpage

### PR DESCRIPTION
**Pull request by SeanChao**
## 该 PR 相关 Issue / Involved issue

Close #

No involved issue

## 完整路由地址 / Example for the proposed route(s)

[https://rsshub.app/sjtu/jwc](https://rsshub.app/sjtu/jwc)

```routes
/sjtu/jwc
/sjtu/jwc/all
/sjtu/jwc/news
/sjtu/jwc/notice
/sjtu/jwc/operation
/sjtu/jwc/affairs
/sjtu/jwc/yjb
/sjtu/jwc/jgb
/sjtu/jwc/zhb
/sjtu/jwc/language
/sjtu/jwc/party
/sjtu/jwc/ge
```

## 新RSS检查列表 / New RSS Script Checklist

- [ ] 这个PR中包含了新的路由吗? Does this PR add new route?
  - 如果有, 请完成检查列表. If yes, please finish the check list
  - **如果你的PR符合下方某个事项, 也请注明. If any of the checklist item meets your PR, please fill it out.**
  - [x] <- 这样打勾
- [x] 是否提供了文档? Documentation provided?
  - [ ] 是否提供了英文文档? EN Documentation provided?
- [x] 是否支持全文获取? Is this RSS Script support fulltext?
  - [x] 如果全文获取中需要访问文章链接, 是否使用了缓存? If fulltext requires to fetch detail pages, is cache used in the process?
  - [缓存说明](https://docs.rsshub.app/joinus/#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun) | [How to use cache](https://docs.rsshub.app/joinus/#ti-jiao-xin-de-rsshub-gui-ze-bian-xie-jiao-ben-shi-yong-huan-cun)
- [ ] 目标是否有明显的反爬/频率限制? Is there any sign of anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? (延长缓存时间, 写文档说明, etc.) If yes, do your code reflect this sign? (e.g. write documentations, use long cache time)
- [ ] 是否引入的新的包? Any new package introduced?
  - 如果有, 请说明原因. If yes, please state your reason
- [ ] 是否使用了`Puppeteer`? Make use of `Puppeteer`?
  - 如果有, 请说明原因. If yes, please state your reason
  

## 说明 / Note

Update rules since the target webpage is updated.

### CODE REVIEW

File: docs/university.md [hunk 1]
```diff
@@ -1484,13 +1484,13 @@ type 列表：
...
 
-<Route author="SeanChao" example="/sjtu/jwc/students" path="/sjtu/jwc/:type?" :paramsDesc="['默认为 students ']">
+<Route author="SeanChao" example="/sjtu/jwc/" path="/sjtu/jwc/:type?" :paramsDesc="['默认为 all ']">
 
-| 面向学生的通知 | 新闻中心 | 通知通告 | 教学运行  | 注册学务 | 研究办 | 教改办 | 综合办 | 工会与支部 |
-| -------------- | -------- | -------- | --------- | -------- | ------ | ------ | ------ | ---------- |
-| students       | news     | notice   | operation | affairs  | yjb    | jgb    | zhb    | party      |
+| 新闻中心 | 通知通告 | 教学运行  | 注册学务 | 研究办 | 教改办 | 综合办 | 语言文字 | 工会与支部 | 通识教育 |
+| -------- | -------- | --------- | -------- | ------ | ------ | ------ | -------- | ---------- | -------- |
+| news     | notice   | operation | affairs  | yjb    | jgb    | zhb    | language | party      | ge       |
```
Comment (NeverBehave) [inline docs/university.md:1493]: 请在正文部分列出所有可能type进行测试

**Review by NeverBehave — CHANGES_REQUESTED** [review]

File: docs/university.md (hunk 1 above)
Comment (SeanChao) [reply docs/university.md:1493]: 加上了

Comment (dzx-dzx) [reply docs/university.md:1493]: @NeverBehave

**Comment by skyzh** [comment]
This PR may close https://github.com/DIYgod/RSSHub/issues/7392

File: lib/routes/universities/sjtu/jwc.js [hunk 2]
```diff
@@ -1,102 +1,97 @@
...
-                const pubDate = new Date(date.getTime() - 60 * 60 * 1000 * (timeZone + serverOffset)).toUTCString();
+                const info = $(e).find('.wz');
+                const relativeLink = info.find('a').attr('href');
+                const link = new URL(relativeLink, sectionLink).href;
+                const title = info.find('a > h2').text();
+                const timeElement = $(e).find('.sj');
+                const day = timeElement.find('h2').text();
+                const yearAndMonth = timeElement.find('p').text();
+                const date = new Date(`${yearAndMonth}.${day}`);
+                const pubDate = new Date(date.getTime() - 60 * 60 * 1000 * (8 + date.getTimezoneOffset() / 60)).toUTCString();
```
Comment (TonyRL) [inline lib/routes/universities/sjtu/jwc.js:94]: Please use parseDate()
docs: https://docs.rsshub.app/joinus/pub-date.html

File: lib/routes/universities/sjtu/jwc.js [hunk 3]
```diff
@@ -1,102 +1,97 @@
...
+        const relativeLink = $(e).attr('href');
+        const absLink = new URL(relativeLink, urlRoot).href;
+        $(e).attr('href', absLink);
+    });
+    return content.html();
 }
 
 module.exports = async (ctx) => {
-    const type = ctx.params.type || 'students';
+    const type = ctx.params.type || 'all';
```
Comment (TonyRL) [inline lib/routes/universities/sjtu/jwc.js:25]:
```suggestion
    const type = ctx.params.type ?? 'all';
```

**Review by TonyRL — CHANGES_REQUESTED** [review]
v2 script standard https://github.com/DIYgod/RSSHub/discussions/8236

**Pull request merged**